src/language-js/parser-babel.js

```diff
@@ -5,9 +5,9 @@ const { hasPragma } = require("./pragma");
 const locFns = require("./loc");
 const postprocess = require("./postprocess");
 
-function babelOptions(extraPlugins = []) {
+function babelOptions({ sourceType, extraPlugins = [] }) {
   return {
-    sourceType: "module",
```

Comment:
Oldest line of code, since first commit https://github.com/vjeux/prettier/commit/29627ee520307494c45073701b20ef64effd3a8f#diff-86100135effef64b185e301e9346e528R13